### PR TITLE
Help Center: Manage notifications in icon and minimized

### PR DESCRIPTION
**Pull request by heavyweight**
Fixes https://github.com/Automattic/dotcom-forge/issues/9341

This PR calculates the unread count on load and based on that the appropriate help center icon is displayed.
Also when the help center is minimized the unread count is displayed and the header is styled differently in this state.

| Before  | After |
| ------------- | ------------- |
| Screenshot 2024-10-29 at 13 02 18  |  Screenshot 2024-10-29 at 13 02 04 |
| Screenshot 2024-10-29 at 13 02 32 | Screenshot 2024-10-29 at 13 02 50|

## Proposed Changes

* Utilize `unreadCount` and set it based on the unread conversations
* Style the help center differently based on this
* Style the help center icon differently based on this

## Testing Instructions
- Test both in calypso and wp-admin
- Create conversations with unread messages
- Check if the appropriate icon is displayed based on this
- Check the header when the help center is minimized. It should be styled differently based on the figma

*

### CODE REVIEW

File: packages/help-center/src/components/utils.tsx [hunk 1]
```diff
@@ -14,3 +15,19 @@ export const generateContactOnClickEvent = (
 		} );
 	}
 };
+
+export const calculateUnread = ( conversations: ZendeskConversation[] ) => {
```
Comment (escapemanuele) [inline packages/help-center/src/components/utils.tsx:19]: Maybe we can use this utility also in https://github.com/Automattic/wp-calypso/blob/trunk/packages/help-center/src/components/help-center-chat-history.tsx#L39-L41 the chat history?

**Comment by escapemanuele** [comment]
I see that with 3 messages in 2 chats, I still see 1 unread when minimized

image
image

**Comment by agrullon95** [comment]
The unread notifications are working correctly in Calypso. I'm able to the see the current numbers of messages and the chats with new messages. I'm also able to see the correct number of conversations with unread messages when I minimize Help Center.

@heavyweight - How can I enable the feature flag on wp-admin? I added it as a URL param, but its not working.

File: packages/help-center/src/components/utils.tsx (hunk 1 above)
Comment (heavyweight) [reply packages/help-center/src/components/utils.tsx:19]: done

**Comment by heavyweight** [comment]
> I see that with 3 messages in 2 chats, I still see 1 unread when minimized
> 
> image image

I sync the store values across the ui now with the last commit, so this should be ok now.
I have issues with the smooch message listeners, they stop working after a while and its tricky to debug them.
I would leave the realtime unread update for a later iteration if needed.

**Comment by heavyweight** [comment]
> The unread notifications are working correctly in Calypso. I'm able to the see the current numbers of messages and the chats with new messages. I'm also able to see the correct number of conversations with unread messages when I minimize Help Center.
> 
> @heavyweight - How can I enable the feature flag on wp-admin? I added it as a URL param, but its not working.

I didn't have issues with this.
I ran `yarn dev --sync` for the hc app, sandbox a simple site and visit `https://zisimplesite.wordpress.com/wp-admin/edit.php?flags=help-center-experience`

**Review by escapemanuele — APPROVED** [review]
Looks great now!

Do we have to add the changes to `help-center-gutenberg.js` too?

**Review by arcangelini — APPROVED** [review]
Everything worked as expected.

One issue I did notice was not caused by this PR but something we should address in the next PR would be the minimized title. It is missing in certain areas like History and Recent conversation.

Screen Shot 2024-10-31 at 12 18 13

**Comment by escapemanuele** [comment]
> One issue I did notice was not caused by this PR but something we should address in the next PR would be the minimized title. It is missing in certain areas like History and Recent conversation.

Added it in another PR @arcangelini

**Pull request merged**